packages/notifier/test/test-notifier-adaptor.js

```diff
@@ -136,3 +140,211 @@ test('notifier adaptor - update from notifier fails', t => {
   return observeNotifier(n, u);
 });
+
+// /////////////////////////////// makeNotifierFromSubscriber /////////////////////////////////
+
+/** @param {{conclusionMethod: 'finish' | 'fail', conclusionValue: any}} config */
+const makeGeometricPublishKit = ({ conclusionMethod, conclusionValue }) => {
+  const { publisher, subscriber } = makeEmptyPublishKit();
+
+  // Publish in geometric batches: [1], [2], [3, 4], [5, 6, 7, 8], ...
+  let nextValue = 1;
+  let nextBatchSize = 1;
+  let done = false;
+  const initialize = () => {
+    publisher.publish(nextValue);
+    nextValue += 1;
+  };
+  const publishNextBatch = () => {
+    if (done) {
+      return;
+    }
+    for (let i = 0; i < nextBatchSize; i += 1) {
+      publisher.publish(nextValue);
+      nextValue += 1;
+    }
+    nextBatchSize *= 2;
+    if (nextValue > 64) {
+      done = true;
+      if (conclusionMethod === 'fail') {
+        publisher.fail(conclusionValue);
+      } else {
+        publisher.finish(conclusionValue);
+      }
+    }
+  };
+
+  return { initialize, publishNextBatch, subscriber };
+};
+
+test('makeNotifierFromSubscriber(finishes) - for-await-of iteration', async t => {
+  const { initialize, publishNextBatch, subscriber } = makeGeometricPublishKit({
+    conclusionMethod: 'finish',
+    conclusionValue: 'done',
+  });
+  const notifier = await makeNotifierFromSubscriber(subscriber);
+  initialize();
+
+  const results = [];
+  for await (const result of notifier) {
+    results.push(result);
+    publishNextBatch();
+  }
+
+  t.deepEqual(
+    results,
+    [1, 2, 4, 8, 16, 32],
+    'only the last of values published between iteration steps should be observed',
+  );
+});
+
+test('makeNotifierFromSubscriber(finishes) - getUpdateSince', async t => {
+  const { initialize, publishNextBatch, subscriber } = makeGeometricPublishKit({
+    conclusionMethod: 'finish',
+    conclusionValue: 'done',
+  });
+  const notifier = await makeNotifierFromSubscriber(subscriber);
+  initialize();
+
+  const results = [];
+  let updateCount;
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    // eslint-disable-next-line no-await-in-loop
+    const result = await notifier.getUpdateSince(updateCount);
+    ({ updateCount } = result);
+    results.push(result.value);
+    // eslint-disable-next-line no-await-in-loop
+    t.deepEqual(await notifier.getUpdateSince(), result);
+    if (updateCount === undefined) {
+      break;
+    }
+    // eslint-disable-next-line no-await-in-loop
+    await publishNextBatch();
+    // eslint-disable-next-line no-await-in-loop
+    t.deepEqual(await notifier.getUpdateSince(), result);
+  }
+
+  t.deepEqual(
+    results,
+    [1, 2, 4, 8, 16, 32, 'done'],
+    'only the last of values published between iteration steps should be observed',
+  );
+  t.deepEqual(await notifier.getUpdateSince(), {
+    value: 'done',
+    updateCount: undefined,
+  });
+});
+
+// TODO: makeNotifierFromSubscriber(fails)
```

Comment:
In general,
```suggestion
test.todo('makeNotifierFromSubscriber(fails)')
```
But this will be done before merging?

Reply:
Yes, done now.